sbncode/SBNEventWeight/App/SystToolsEventWeight_module.cc

```diff
@@ -0,0 +1,238 @@
+#include "canvas/Persistency/Common/Assns.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+
+#include "sbnobj/Common/SBNEventWeight/EventWeightMap.h"
+#include "sbnobj/Common/SBNEventWeight/EventWeightParameterSet.h"
+
+#include "systematicstools/interface/ISystProviderTool.hh"
+#include "systematicstools/interpreters/ParamHeaderHelper.hh"
+#include "systematicstools/utility/ParameterAndProviderConfigurationUtility.hh"
+#include "systematicstools/utility/exceptions.hh"
+
+#include "canvas/Persistency/Common/Assns.h"
+#include "art/Framework/Principal/Run.h"
+
+namespace sbn {
+  namespace evwgh {
+
+class SystToolsEventWeight : public art::EDProducer {
+public:
+  explicit SystToolsEventWeight(fhicl::ParameterSet const& p);
+
+  SystToolsEventWeight(SystToolsEventWeight const &) = delete;
+  SystToolsEventWeight(SystToolsEventWeight &&) = delete;
+  SystToolsEventWeight& operator = (SystToolsEventWeight const&) = delete;
+  SystToolsEventWeight& operator = (SystToolsEventWeight&&) = delete;
+
+private:
+  void produce(art::Event& e) override;
+  void beginRun(art::Run& run) override;
+
+private:
+  std::string fGenieModuleLabel;
+  bool fAllowMissingTruth;
+  bool fDebugMode;
+
+  systtools::provider_list_t fSystProviders;
+  systtools::param_header_map_t fParamHeaderMap;
+  systtools::ParamHeaderHelper fParamHeaderHelper;
+
+};
+
+
+SystToolsEventWeight::SystToolsEventWeight(fhicl::ParameterSet const& p)
+  : EDProducer{p}
+{
+
+  fGenieModuleLabel = p.get<std::string>("generator_module_label", "generator");
+  fAllowMissingTruth = p.get<bool>("AllowMissingTruth");
+  fDebugMode = p.get<bool>("debugMode");
+
+  fhicl::ParameterSet syst_provider_config = p.get<fhicl::ParameterSet>("generated_systematic_provider_configuration");
+
+  MF_LOG_INFO("SystToolsEventWeight") << "Configuring ISystProvider";
+
+  fSystProviders = systtools::ConfigureISystProvidersFromParameterHeaders(syst_provider_config);
+  fParamHeaderMap = systtools::BuildParameterHeaders(fSystProviders);
+  fParamHeaderHelper = systtools::ParamHeaderHelper(fParamHeaderMap);
+
+  MF_LOG_INFO("SystToolsEventWeight") << "ISystProvider is configuered" << std::endl;
+
+  produces<std::vector<sbn::evwgh::EventWeightMap> >();
+  produces<art::Assns<simb::MCTruth, sbn::evwgh::EventWeightMap> >();
+  produces<std::vector<sbn::evwgh::EventWeightParameterSet>, art::InRun>();
+
+}
+
+
+void SystToolsEventWeight::produce(art::Event& e) {
+
+  auto mcwghvec = std::make_unique<std::vector<sbn::evwgh::EventWeightMap> >();
+  auto wghassns = std::make_unique<art::Assns<simb::MCTruth, sbn::evwgh::EventWeightMap> >();
+
+  art::PtrMaker<sbn::evwgh::EventWeightMap> makeWeightPtr(e);
+
+  std::vector<art::Ptr<simb::MCTruth> > mclist;
+  art::Handle<std::vector<simb::MCTruth>> mcTruthHandle;
+  if(!fGenieModuleLabel.empty()) e.getByLabel(fGenieModuleLabel, mcTruthHandle);
```

Comment:
`getByLabel()` is being deprecated by _art_:
```suggestion
  if(!fGenieModuleLabel.empty()) mcTruthHandle = e.getHandle<std::vector<simb::MCTruth>>(fGenieModuleLabel);
```

Reply:
Updated as suggested.